### PR TITLE
PEM-3896-Added content for changing agent.cfg file path in pemagent service file

### PR DESCRIPTION
**Pull request by arup-roy2023**
## What Changed?

Added the following sub-heading in the "Registering a PEM agent"- "Modifying the pemagent service file to update agent configuration file path" for steps to update the service file

### CODE REVIEW

File: product_docs/docs/pem/9/registering_agent.mdx [hunk 1]
```diff
@@ -233,3 +233,31 @@ To use a nonroot user account to register a PEM agent, you must first install th
...
 !!! Note
     Any probes and jobs that require root permission or access to a file owned by another user (for example,  enterprisedb) fail.
+
+## Modifying the pemagent service file to update agent configuration file path
+
+In some situations, you might need to move the agent configuration file from its default location `/usr/edb/pem/agent/etc/agent.cfg` to another location. In that case, you need to update the value of the agent configuration file path in the `pemagent` service file. 
+
+1. Use the following command to modify the `pemagent` service file as a superuser:
+
+    ```ini
```
Comment (sonotley) [inline product_docs/docs/pem/9/registering_agent.mdx:243]:
```suggestion
    ```shell
```

File: product_docs/docs/pem/9/registering_agent.mdx [hunk 2]
```diff
@@ -233,3 +233,31 @@ To use a nonroot user account to register a PEM agent, you must first install th
...
+    $ sudo vi  /usr/lib/systemd/system/pemagent.service
+    ```
+
+2. Update the value for the `agent.cfg` file path. For example, on a CentOS 7.x host update the value in the following line: 
+
+    ```ini
```
Comment (sonotley) [inline product_docs/docs/pem/9/registering_agent.mdx:249]:
```suggestion
    ```shell
```

File: product_docs/docs/pem/9/registering_agent.mdx [hunk 3]
```diff
@@ -233,3 +233,31 @@ To use a nonroot user account to register a PEM agent, you must first install th
...
     Any probes and jobs that require root permission or access to a file owned by another user (for example,  enterprisedb) fail.
+
+## Modifying the pemagent service file to update agent configuration file path
+
+In some situations, you might need to move the agent configuration file from its default location `/usr/edb/pem/agent/etc/agent.cfg` to another location. In that case, you need to update the value of the agent configuration file path in the `pemagent` service file. 
+
+1. Use the following command to modify the `pemagent` service file as a superuser:
+
+    ```ini
+    $ sudo vi  /usr/lib/systemd/system/pemagent.service
```
Comment (sonotley) [inline product_docs/docs/pem/9/registering_agent.mdx:244]: I thought we weren't supposed to include the prompt in code blocks?

Comment (nidhibhammar) [reply product_docs/docs/pem/9/registering_agent.mdx:244]: yes, we should not use prompts in the code blocks. @arup-roy2023, please remove the prompt signs from all the code blocks

File: product_docs/docs/pem/9/registering_agent.mdx [hunk 4]
```diff
@@ -233,3 +233,31 @@ To use a nonroot user account to register a PEM agent, you must first install th
 
 !!! Note
     Any probes and jobs that require root permission or access to a file owned by another user (for example,  enterprisedb) fail.
+
+## Modifying the pemagent service file to update agent configuration file path
```
Comment (nidhibhammar) [inline product_docs/docs/pem/9/registering_agent.mdx:237]: I think this should go in the Troubleshooting section. We can provide the context that if you run into this situation, go to the troubleshooting section and provide a link to that section. Also, I think the heading should be describing what the problem is. This heading sounds more like a solution to the problem.

File: product_docs/docs/pem/9/registering_agent.mdx (hunk 3 above)
Comment (nidhibhammar) [inline product_docs/docs/pem/9/registering_agent.mdx:244]:
```suggestion
    # Running as a superuser
    sudo vi  /usr/lib/systemd/system/pemagent.service
```

Comment (nidhibhammar) [reply product_docs/docs/pem/9/registering_agent.mdx:244]: Same suggestion for all the code blocks

File: product_docs/docs/pem/9/registering_agent.mdx [hunk 5]
```diff
@@ -233,3 +233,31 @@ To use a nonroot user account to register a PEM agent, you must first install th
...
+    $ sudo vi  /usr/lib/systemd/system/pemagent.service
+    ```
+
+2. Update the value for the `agent.cfg` file path. For example, on a CentOS 7.x host update the value in the following line: 
+
+    ```ini
+    ExecStart=/usr/edb/pem/agent/bin/pemagent -c /usr/edb/pem/agent/etc/agent.cfg
+    ```
+
+3. Reload `systemd`, updating the modified service scripts:
```
Comment (nidhibhammar) [inline product_docs/docs/pem/9/registering_agent.mdx:253]:
```suggestion
3. Reload `systemd`, updating the modified service scripts:
```
```suggestion
3. Reload `systemd`, to update the modified service script:
```

**Review by nidhibhammar — CHANGES_REQUESTED** [review]
I have given my review comments.

File: product_docs/docs/pem/9/registering_agent.mdx (hunk 1 above)
Comment (arup-roy2023) [reply product_docs/docs/pem/9/registering_agent.mdx]: Changed.

File: product_docs/docs/pem/9/registering_agent.mdx (hunk 2 above)
Comment (arup-roy2023) [reply product_docs/docs/pem/9/registering_agent.mdx]: Changed

File: product_docs/docs/pem/9/registering_agent.mdx (hunk 3 above)
Comment (arup-roy2023) [reply product_docs/docs/pem/9/registering_agent.mdx]: I have removed the prompt symbol from all the code blocks.

File: product_docs/docs/pem/9/registering_agent.mdx (hunk 4 above)
Comment (arup-roy2023) [reply product_docs/docs/pem/9/registering_agent.mdx]: I have added a note in the "Registering a PEM agent" topic indicating the situation/issue the user might face and provided the link to the "Troubleshooting" topic, where the steps have been added. The topic heading has also been changed.

File: product_docs/docs/pem/9/registering_agent.mdx (hunk 3 above)
Comment (arup-roy2023) [reply product_docs/docs/pem/9/registering_agent.mdx]: Updated

File: product_docs/docs/pem/9/registering_agent.mdx [hunk 6]
```diff
@@ -233,3 +233,31 @@ To use a nonroot user account to register a PEM agent, you must first install th
...
+
+In some situations, you might need to move the agent configuration file from its default location `/usr/edb/pem/agent/etc/agent.cfg` to another location. In that case, you need to update the value of the agent configuration file path in the `pemagent` service file. 
+
+1. Use the following command to modify the `pemagent` service file as a superuser:
+
+    ```ini
+    $ sudo vi  /usr/lib/systemd/system/pemagent.service
+    ```
+
+2. Update the value for the `agent.cfg` file path. For example, on a CentOS 7.x host update the value in the following line:
```
[the remark this answers is not in the document]
Comment (arup-roy2023) [reply product_docs/docs/pem/9/registering_agent.mdx]: Changed the line

File: product_docs/docs/pem/9/registering_agent.mdx (hunk 5 above)
Comment (arup-roy2023) [reply product_docs/docs/pem/9/registering_agent.mdx]: Updated the line

**Comment by arup-roy2023** [comment]
I have updated two topics: "Registering a PEM agent" and "Troubleshooting", based on the review comments. Please check the changes.

File: product_docs/docs/pem/9/troubleshooting.mdx [hunk 7]
```diff
@@ -88,3 +88,32 @@ In some situations, you might need to uninstall the PEM server, reinstall it, an
     ```shell
     /usr/edb/pem/bin/configure-pem-server.sh
     ```
+
+## Updating pemagent service file values if agent configuration file path is modified
```
Comment (nidhibhammar) [inline product_docs/docs/pem/9/troubleshooting.mdx:92]: I think we can create separate child topics for the server and agent under the troubleshooting topic. Also, the heading for this need to address the problem and not the solution.

File: product_docs/docs/pem/9/registering_agent.mdx [hunk 8]
```diff
@@ -219,45 +219,18 @@ To use a nonroot user account to register a PEM agent, you must first install th
...
-    ```
-
-4. Restart the PEM agent:
-
-    ```shell
-    $ sudo systemctl restart pemagent
-    ```
+!!! Note 
+    - Any probes and jobs that require root permission or access to a file owned by another user (for example,  enterprisedb) fail.
+    - If you move the `agent.cfg` file from its default location to another, the agent status on the PEM dashboard may display an error; see [Updating pemagent service file values if agent configuration file path is modified](/pem/latest/troubleshooting/#updating-pemagent-service-file-values-if-agent-configuration-file-path-is-modified).
```
Comment (nidhibhammar) [inline product_docs/docs/pem/9/registering_agent.mdx:236]: I think the agent status shows up "unknown"

File: product_docs/docs/pem/9/troubleshooting.mdx (hunk 7 above)
Comment (arup-roy2023) [reply product_docs/docs/pem/9/troubleshooting.mdx]: Based on the last discussion with Nidhi, two child topics, one for the server and another for the agent, are not required right now in the "Troubleshooting" topic. 
I have modified the heading to highlight the problem: "Agent status on the PEM dashboard is displayed as unknown"

**Review by drothery-edb — APPROVED** [review]

**Pull request merged**